mathesar/api/serializers/tables.py

```diff
@@ -67,10 +72,38 @@ def get_previews_url(self, obj):
 
     def validate_data_files(self, data_files):
         if data_files and len(data_files) > 1:
-            raise ValidationError('Multiple data files are unsupported.')
+            raise MultipleDataFileAPIException()
         return data_files
 
+    def create(self, validated_data):
+        schema = validated_data['schema']
+        data_files = validated_data.get('data_files')
+        name = validated_data.get('name') or gen_table_name(schema, data_files)
 
-class TablePreviewSerializer(serializers.Serializer):
+        try:
+            if data_files:
+                table = create_table_from_datafile(data_files, name, schema)
+            else:
+                table = create_empty_table(name, schema)
+        except ProgrammingError as e:
+            if type(e.orig) == DuplicateTable:
+                raise DuplicateTableAPIException(e, message=f"Relation {validated_data['name']}"
+                                                            f" already exists in schema {schema.id}",
+                                                 field="name", status_code=status.HTTP_400_BAD_REQUEST)
```

Comment:
```suggestion
                raise DuplicateTableAPIException(
                    e,
                    message=f"Relation {validated_data['name']} already exists in schema {schema.id}",
                    field="name",
                    status_code=status.HTTP_400_BAD_REQUEST,
                )
```
This has the added benefit of making it more reasonable to avoid splitting the message across two lines.